pkg/api/auth.go

```diff
@@ -20,46 +25,89 @@ func authFail(w http.ResponseWriter, realm string, delay int) {
 	WriteJSON(w, http.StatusUnauthorized, NewError(UNAUTHORIZED))
 }
 
+// nolint (gocyclo) - we use closure making this a complex subroutine
 func BasicAuthHandler(c *Controller) mux.MiddlewareFunc {
 	realm := c.Config.HTTP.Realm
 	if realm == "" {
 		realm = "Authorization Required"
 	}
 	realm = "Basic realm=" + strconv.Quote(realm)
-	delay := c.Config.HTTP.Auth.FailDelay
 
-	if c.Config.HTTP.Auth.HTPasswd.Path == "" {
+	// no password based authN, if neither LDAP nor HTTP BASIC is enabled
+	if c.Config.HTTP.Auth == nil || (c.Config.HTTP.Auth.HTPasswd.Path == "" && c.Config.HTTP.Auth.LDAP == nil) {
```

Comment:
@hallyn either c.Config.HTTP.Auth is nil which means it enters and returns, or FailDelay is not specified and defaults to 0 (nil value for int type) but it is otherwise configurable.